File: Intermediate/Topic-001-Introduction-to-Problem-Solving/Assignment/Q1-Count-Factors/count-factor.py
import sys

def count_factors(n):
    if n <= 0:
        return 0
    
    count = 0


    for i in range (1, int(n**0.5) + 1):
        if n % i == 0:
            # Perfect square case
            if i * i == n:
                count += 1
            else:
                count += 2 

    return count

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python count-factor.py <positive_integer>")
        sys.exit(1)

    try:
        number = int(sys.argv[1])
        if number <= 0:
            raise ValueError
    except ValueError:
        print("Please provide a valid positive integer.")
        sys.exit(1)

    factors_count = count_factors(number)
    print(f"The number of factors of {number} is: {factors_count}")

File: Intermediate/Topic-001-Introduction-to-Problem-Solving/Assignment/Q1-Count-Factors/count-factor.c
#include<stdio.h>

int count_factors(int n);



int main() {
    int number;
    printf("Enter the positive integer: ");
    scanf("%d", &number);

    int num_factors = count_factors(number);
    printf("Number of factors of %d is %d\n", number, num_factors);
    
    return 0;
}


int count_factors(int n) {
    int count = 0;
    for (int i = 1; i * i <= n; i++) {
        if (n%i == 0) {
            // Perfect square case
            if (i * i == n) {
                count++;
            } else {
                count += 2;
            }
        }
    }
    return count;

}